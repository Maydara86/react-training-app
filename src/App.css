body {
  margin: 0;
  height: 100%;
  padding: 10px;
}

.articleSummaryGridContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-auto-columns: 1fr 1fr 1fr;
  grid-gap: 1em 1em;
  place-items: center center;
  place-content: center center;
}

.ListContainer {
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10px;
}

.ListItem:nth-child(1) {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}