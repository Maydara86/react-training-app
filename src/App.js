import React from 'react'
import './App.css'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import ArticleSummaryListItem from './components/Article/ArticlesList/ArticleSummaryListItem'
import clickStarHandler, { clickFollowHandler } from './store/actions/usersAction'
import clickBookmarkHandler, { clickClapHandler } from './store/actions/articlesAction'
import usersSelector from './selectors/users'
import articlesSelector from './selectors/articles'

function App(props) {
  /* eslint-disable no-shadow */
  const {
    users,
    articles,
    clickClapHandler,
    clickBookmarkHandler,
    clickStarHandler,
    clickFollowHandler,
  } = props
  /* eslint-disable no-shadow */

  return (
    <div>
      <div className="ListContainer">
        {articles.map((article, i) => {
          return (
            <ArticleSummaryListItem
              className="ListItem"
              key={article.id}
              {...article}
              user={users[i]}
              handleClapClick={clickClapHandler}
              handleBookmarkClick={clickBookmarkHandler}
              handleFollowClick={clickFollowHandler}
              handleStarClick={clickStarHandler}
            />
          )
        })}
      </div>
    </div>
  )
}

const mapStateToProps = state => {
  return {
    articles: articlesSelector(state),
    users: usersSelector(state),
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    { clickStarHandler, clickBookmarkHandler, clickClapHandler, clickFollowHandler },
    dispatch
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
