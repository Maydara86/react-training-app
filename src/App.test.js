import React from 'react'
import { shallow } from './enzyme'
import App from './App'

jest.mock('./data/users-data.json')
const { users } = require('./data/users-data.json')

const AppComponent = shallow(<App />)
const wrapper = AppComponent.instance()

describe('App Component', () => {
  it('testing clickFollowHandler method to have the expected effect on the state of the first user', () => {
    wrapper.clickFollowHandler('5d552d0058f193f2795fc814')
    expect(wrapper.state.users[0].isFollowed).toBeFalsy()
  })

  it('clickFollowHandler method have the expected effect on the state of the second user', () => {
    wrapper.clickFollowHandler('5d552d00b20b141dff10d2a2')
    expect(wrapper.state.users[1].isFollowed).toBeFalsy()
  })

  it('Testing clickStarHandler method to have the expected effect on the state of the first user', () => {
    wrapper.clickStarHandler('5d552d0058f193f2795fc814')
    expect(wrapper.state.users[0].isStarred).toBeFalsy()
  })

  it('calling the clickStarHandler method has the expected effect on the state of the second user', () => {
    wrapper.clickStarHandler('5d552d00b20b141dff10d2a2')
    expect(wrapper.state.users[1].isStarred).toBeFalsy()
  })

  it('checks the clickClapHandler method', () => {
    wrapper.clickClapHandler('5d66ae9445543ffea5167d5e')
    expect(wrapper.state.articles[1].claps).toEqual(71)
    expect(wrapper.state.articles[1].didClap).toBeTruthy()
  })

  it('checks the clickBookmarkHandler method', () => {
    wrapper.clickBookmarkHandler('5d66ae9445543ffea5167d5e')
    expect(wrapper.state.articles[1].bookmark).toBeTruthy()
  })
})
