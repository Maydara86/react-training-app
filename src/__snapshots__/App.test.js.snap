// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <div
    className="ListContainer"
  >
    <div
      className="articleContainer"
    >
      <div
        className="userContainer"
      >
        <img
          alt="Walton Morton"
          className="userImageItem"
          src="avatar.png"
        />
        <div
          className="userInfoItem"
        >
          <div
            className="contentContainerRow1"
          >
            <div
              className="nameItem"
            >
              Walton Morton
            </div>
          </div>
          <div
            className="contentContainerRow2"
          >
            <div
              className="dateItem"
            >
              Aug 9
            </div>
            <span>
              Â·
            </span>
            <div
              className="readingTimeItem"
            >
              20
               min read
            </div>
            <svg
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fill": "grey",
                  "height": 15,
                  "marginTop": 4,
                  "padding": 4,
                  "width": 15,
                }
              }
              viewBox="0 0 15 15"
            >
              <path
                d="M7.44 2.32c.03-.1.09-.1.12 0l1.2 3.53a.29.29 0 00.26.2h3.88c.11 0 .13.04.04.1L9.8 8.33a.27.27 0 00-.1.29l1.2 3.53c.03.1-.01.13-.1.07l-3.14-2.18a.3.3 0 00-.32 0L4.2 12.22c-.1.06-.14.03-.1-.07l1.2-3.53a.27.27 0 00-.1-.3L2.06 6.16c-.1-.06-.07-.12.03-.12h3.89a.29.29 0 00.26-.19l1.2-3.52z"
              />
            </svg>
          </div>
        </div>
      </div>
      <img
        alt="amet aliqua cupidatat dolore Lorem elit voluptate proident tempor incididunt"
        className="articleImageItem"
        src="article.jpeg"
      />
      <h3
        className="articleTitleItem"
      >
        amet aliqua cupidatat dolore Lorem elit voluptate proident tempor incididunt
      </h3>
      <p />
      <span
        className="readMore"
      >
        Read more...
      </span>
      <div
        className="appreciation"
      >
        <div
          className="claps"
        >
          <svg
            className="clap"
            height="25"
            onClick={[Function]}
            viewBox="0 0 25 25"
            width="25"
          >
            <g
              fillRule="evenodd"
            >
              <path
                d="M11.74 0l.76 2.97.76-2.97zM14.81 3.78l1.84-2.56-1.42-.47zM8.38 1.22l1.84 2.56L9.8.75zM20.38 21.62a5.11 5.11 0 0 1-3.16 1.61l.49-.45c2.88-2.89 3.45-5.98 1.69-9.21l-1.1-1.94-.96-2.02c-.31-.67-.23-1.18.25-1.55a.84.84 0 0 1 .66-.16c.34.05.66.28.88.6l2.85 5.02c1.18 1.97 1.38 5.12-1.6 8.1M7.1 21.1l-5.02-5.02a1 1 0 0 1 .7-1.7 1 1 0 0 1 .72.3l2.6 2.6a.44.44 0 0 0 .63-.62L4.1 14.04l-1.75-1.75a1 1 0 1 1 1.41-1.41l4.15 4.15a.44.44 0 0 0 .63 0 .44.44 0 0 0 0-.62L4.4 10.26 3.22 9.08a1 1 0 0 1 0-1.4 1.02 1.02 0 0 1 1.41 0l1.18 1.16L9.96 13a.44.44 0 0 0 .62 0 .44.44 0 0 0 0-.63L6.43 8.22a.99.99 0 0 1-.3-.7.99.99 0 0 1 .3-.7 1 1 0 0 1 1.41 0l7 6.98a.44.44 0 0 0 .7-.5l-1.35-2.85c-.31-.68-.23-1.19.25-1.56a.85.85 0 0 1 .66-.16c.34.06.66.28.88.6L18.63 14c1.57 2.88 1.07 5.54-1.55 8.16a5.62 5.62 0 0 1-5.06 1.65 9.35 9.35 0 0 1-4.93-2.72zM11 5.98l2.56 2.56c-.5.6-.56 1.41-.15 2.28l.26.56-4.25-4.25a.98.98 0 0 1-.12-.45 1 1 0 0 1 .29-.7 1.02 1.02 0 0 1 1.41 0zm8.89 2.06c-.38-.56-.9-.92-1.49-1.01a1.74 1.74 0 0 0-1.34.33c-.38.29-.61.65-.71 1.06a2.1 2.1 0 0 0-1.1-.56 1.78 1.78 0 0 0-.99.13l-2.64-2.64a1.88 1.88 0 0 0-2.65 0 1.86 1.86 0 0 0-.48.85 1.89 1.89 0 0 0-2.67-.01 1.87 1.87 0 0 0-.5.9c-.76-.75-2-.75-2.7-.04a1.88 1.88 0 0 0 0 2.66c-.3.12-.61.29-.87.55a1.88 1.88 0 0 0 0 2.66l.62.62a1.88 1.88 0 0 0-.9 3.16l5.01 5.02c1.6 1.6 3.52 2.64 5.4 2.96a7.16 7.16 0 0 0 1.18.1c1.03 0 2-.25 2.9-.7A5.9 5.9 0 0 0 21 22.24c3.34-3.34 3.08-6.93 1.74-9.17l-2.87-5.04z"
              />
            </g>
          </svg>
          <span
            className="clapsNumber"
          >
            46
          </span>
        </div>
        <div
          className="responses"
        >
          <div
            className="responsesNumber"
          >
             responses
          </div>
          <svg
            className="bookmark"
            height="25"
            onClick={[Function]}
            viewBox="0 0 25 25"
            width="25"
          >
            <path
              d="M19 6a2 2 0 0 0-2-2H8a2 2 0 0 0-2 2v14.66h.01c.01.1.05.2.12.28a.5.5 0 0 0 .7.03l5.67-4.12 5.66 4.13c.2.18.52.17.71-.03a.5.5 0 0 0 .12-.29H19V6z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
